internal/db/repos/instance.go

```diff
@@ -76,13 +76,32 @@ func (r *InstanceRepository) UpdateStatusByName(ctx context.Context, name string
 		Update("status", status).Error
 }
 
-// List retrieves a paginated list of instances
+// List retrieves a paginated list of instances with optional status and soft delete filters
 func (r *InstanceRepository) List(ctx context.Context, opts *models.ListOptions) ([]models.Instance, error) {
 	var instances []models.Instance
 	query := r.db.WithContext(ctx)
 
-	if opts != nil && opts.IncludeDeleted {
-		query = query.Unscoped()
+	if opts != nil {
+		// Apply status filter if provided
+		if opts.Status != nil {
+			query = query.Where("status = ?", *opts.Status)
+		} else {
+			// By default, only show non-terminated instances
+			query = query.Where("status != ?", models.InstanceStatusTerminated)
+		}
```

Comment:
The default behavior should be handled in the API layer, not the DB layer.  So if no opts are passed in by the user we should create the opts that are this default no terminated response.

The lower level the function, the less context it should have. 

These DB methods should not do any "thinking", they should just validate the inputs and execute. 

Any context that is put here has ripple effects upwards to everything that might need access to the DB. 

This is a design principle we should apply throughout Talis.